backend/src/routes/getBalance.ts

```diff
@@ -0,0 +1,32 @@
+import { FastifyInstance } from 'fastify';
+
+const addressRegex = /^0x0[0-9a-fA-F]{63}$/;
+
+export function getBalanceRoute(fastify: FastifyInstance) {
+  fastify.get('/get_balance', async (request, reply) => {
+    const { address } = request.query as { address?: string };
+
+    if (!address || !addressRegex.test(address)) {
+      reply.status(400).send({ error: 'Invalid address format' });
+      return;
+    }
+
+    try {
+      const { rows } = await fastify.pg.query<{ balance: string }>(
+        'SELECT balance FROM balance_usdc WHERE address = $1 ORDER BY block_number DESC LIMIT 1',
+        [address],
+      );
+
+      if (rows.length === 0) {
```

Comment:
Shouldn't it return `0` if there's no balance for the address? Imagine an user that just deployed their wallet and so there is no USDC transaction for them, in the current implementation the api returns an error but in the UI we realy just want to show `0` to them.